core/client/fs/src/main/java/alluxio/client/block/stream/NettyPacketWriter.java

```diff
@@ -110,6 +113,9 @@
   /** This condition is met if there is nothing in the netty buffer. */
   private final Condition mBufferEmptyOrFailed = mLock.newCondition();
 
+  /** Flush should not be called before write. */
+  private boolean mWrote = false;
```

Comment:
Could we allow the RPC to go through and no-op on the server side? This would simplify the client state.

Reply:
Now the client side will send the flush command to worker side.
Worker side will not actually flush() if no write() occurred.